python/kserve/test/test_server.py

```diff
@@ -227,6 +251,16 @@ def test_predict(self, http_server_client):
         assert resp.content == b'{"predictions":[[1,2]]}'
         assert resp.headers['content-type'] == "application/json"
 
+    def test_infer(self, http_server_client):
+        input_data = b'{"inputs": [{"name": "input-0","shape": [1, 2],"datatype": "FP32","data": [[1,2]]}]}'
+        resp = http_server_client.post('/v2/models/TestModel/infer',
+                                       data=input_data)
+
+        result = json.loads(resp.content)
+        assert resp.status_code == 200
+        assert result["outputs"][0]["data"] == [[1, 2]]
+        assert resp.headers['content-type'] == "application/json"
```

Comment:
can we move this logic to helper method also? seems like we repeat this starting at line 322